protocol/messenger.go

```diff
@@ -1397,6 +1398,24 @@ func (m *Messenger) watchIdentityImageChanges() {
 	}()
 }
 
+func (m *Messenger) watchPendingCommunityRequestToJoin() {
+	m.logger.Debug("watching community request to join")
+
+	go func() {
+		for {
+			select {
+			case <-time.After(time.Minute):
```

Comment:
Does it do it every minute? maybe we can increase it, since it's not critical to do it that often?

Reply:
Yes, It will check every minute. I and @cammellos initially agreed to do it every minute because if the admin opens the mobile/desktop near the request timeout, a shorter polling time ensures the request is deleted.

Reply:
Is there a way to do it kind of like a `do/while`, where it does it the first time immediately (or even better after a short delay), and then does it every, let's say, 10 minutes?
This way, we ensure to delete the requests at open, but also don't do a check on all of them each minute?
Maybe it's not that big of a deal either TBH. It's just that we have/had performance issues, so limiting DB calls as much as possible is good.

Reply:
I agree we should limit the DB calls. I guess 10 minutes should be fine, given that the timeout is seven days.

We expose the `CheckAndDeletePendingRequestToJoinCommunity` method in `PublicAPI` so the client can call it at the successful login. This would clear any pending requests which are expired/timed out.

I will update the polling to 10 minutes 👍